{
  if (!renameSupported())   return;
  Path src=getTestRootPath(fSys,"test/hadoop/dir");
  fSys.mkdirs(src);
  Path dst=getTestRootPath(fSys,"test/nonExistent/newdir");
  try {
    rename(src,dst,false,true,false,Rename.NONE);
    Assert.fail("Expected exception was not thrown");
  }
 catch (  IOException e) {
    Assert.assertTrue(unwrapException(e) instanceof FileNotFoundException);
  }
  try {
    rename(src,dst,false,true,false,Rename.OVERWRITE);
    Assert.fail("Expected exception was not thrown");
  }
 catch (  IOException e) {
    Assert.assertTrue(unwrapException(e) instanceof FileNotFoundException);
  }
}
